timeline.js

```diff
@@ -108,15 +108,109 @@
     if (self.validate()) {
       // Load library.json - need to inform TimelineJS which version it is
       $.getJSON(self.getLibraryFilePath('library.json'), function (data) {
-        new TimelineJS({
-          type: 'timeline',
+
+        var datas = {
+          "title": {
+            "text": {
+              "headline": self.options.timeline.headline,
+              "text": self.options.timeline.text
+            }
+          },
+          "events": [],
+          "eras": []
+        };
+
+        if (self.options.timeline.asset !== undefined && self.options.timeline.asset.media) {
+          datas.title.media = {
+            "url": self.options.timeline.asset.media,
+            "caption": self.options.timeline.asset.caption,
+            "credit": self.options.timeline.asset.credit,
+            "thumbnail": self.options.timeline.asset.thumbnail
+          };
+        }
+
+        // convert date to TimelineJS3
```

Comment:
You may want to use H5P's ugrades.js functionality to upgrade parameters once when updating content to the new minor version instead of doing that over and over again when the content type runs. I don't think there's a good example, but Interactive Video has quite some history, so this might help as reference: https://github.com/h5p/h5p-interactive-video/blob/master/upgrades.js

Reply:
I though about using upgrade but I don't think it's a good idea here.

The sementics followed exactly the same structure as the object given to timeline, which allow them to just give the params variable to the timeline library.

However the v3 structure is more complex. It would require 6 fields instead of 2 for each date which will might quickly become ugly. There is also some additionnal depth for some option so this would still require reparsing each date. Also there is some additionnal check required because the timelineJS3 library "crash" if some params are defined as undefined.

Reply:
Ah, I see. So there then are some options that Timeline 3 offers that you cannot use with H5P? Hmmm. Guess we'll have to wait and see when someone notices that :-)

Reply:
Yes (https://timeline.knightlab.com/docs/options.html and https://timeline.knightlab.com/docs/json-format.html#json-era).
The display date could solve a request on the forum, but if new functionnality should be added, it should probably be in an other PR aftet this one get merged (if it get merged).